### DISCUSSION TITLE
Android上Termux中运行无法解析DNS：get_dnsserver_list失败，不支持linux-posix平台

### DISCUSSION THREAD

**Discussion by majiayeah** [SeaHOH/GotoX#114] [展示]
首先感谢作者的作品，很好用！

由于本人只需要SNI伪造功能，用来直连一些网站，所以服务器端的GAE部分完全不需要。
一直在Windows系统中用的很好。

最近想移植到Android上使用。没有root。
由于没有apk包，所以需要用一些其它的办法。
选用安装Termux来运行：
1. 安装Termux，在Termux安装python
2. 在python中pip install本软件requirement.txt中的所有依赖包
3. 在Termux运行Dnscrypt-Proxy，监听127.0.0.1:5353 （未root不能使用1024以下端口）
4. 在GotoX中config.ini中，配置DNS server为127.0.0.1:5353
5. 导入根证书
6. Termux中运行python start.py

基本功能都运行良好，除了一点：DNS解析

GotoX页面显示：
12:01 W get_dnsserver_list失败，不支持linux-posix平台
12:01 W 读取系统当前DNS设置失败

目前GotoX无法调用127.0.0.1:5353进行DNS解析，导致仍然被DNS污染。

尝试过使用Daedalus，全局使用127.0.0.1:5353作为DNS服务器。但似乎GotoX不受约束，依然不知道连接到哪里的DNS服务器去了，依然污染。

请问有什么好的办法可以解决Android上的DNS问题吗？
如果能解决，那就能完美运行使用了。

**Comment by SeaHOH** [comment]
现在是固定使用 53 端口，我会修改支持自定义，请等待。

**Comment by majiayeah** [comment]
> 现在是固定使用 53 端口，我会修改支持自定义，请等待。

好的，谢谢！

**Comment by SeaHOH** [comment]
已添加，我这里工作正常，请测试。

**Comment by majiayeah** [comment]
> 已添加，我这里工作正常，请测试。

你好，在哪里下载？release里好像没有新的版本？

**Comment by SeaHOH** [comment]
直接下载 master 源码就行。
